[12.0][MIG] account_asset_management

apps2grow opened the pull request:
TODO:
- Migration script for account.asset parent_path (replacing parent_left and parent_right)
- models/account_fiscal_year.py instead of models/date_range.py (I don't understand the purpose of this logic)

luc-demeyer committed:
add asset management modules

luc-demeyer committed:
asset mgt update

luc-demeyer committed:
redo

luc-demeyer committed:
synch asset mgt with recent V7 changes

rdeheele committed:
[FIX] test floats with assertAlmostEqual

rdeheele committed:
[UPD] add places arg in assertAlmostEqual tests

rdeheele committed:
[UPD] flake clean

lepistone committed:
move account_asset_management* out of __unported__

lepistone committed:
make *asset* installable again

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

Commit:
[FIX] Inherited model is a transient model

sbidoul committed:
[IMP] account_asset_management: use reload_on_button instead of reloading page

Commit:
[IMP][account_asset_management] Define a list of move's fields that can't be modified if a move is linked with a depriciation line

Commit:
[IMP][account_asset_management] Define FIELDS_AFFETCS_ASSET_MOVE as a set directly

Commit:
[IMP][account_asset_management] Typo

luc-demeyer committed:
account_asset_management: README.rst

luc-demeyer committed:
[IMP] account_asset_management: code cosmetics

luc-demeyer committed:
account_asset_management: group invoice lines fix

luc-demeyer committed:
analytic enhancement

Commit:
[ADD][account_asset_management] Add check on account move line

luc-demeyer committed:
analytic enhancement

luc-demeyer committed:
analytic enhancement

Commit:
[FIX][account_asset_management] Add items on affects fields list

luc-demeyer committed:
80 asset mgt analytic fix

sbidoul committed:
[FIX] account_asset_management: do not propose special and closed periods for asset posting

yvaucher committed:
remove old icon

Commit:
[FIX][account_asset_management] Avoid to select closed or cancelled analytic account

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

pedrobaeza committed:
[MIG] Rename manifest files

sbidoul committed:
remove obsolete .pot files [ci skip]

luc-demeyer committed:
[10.0][MIG]account_asset_management suite refactoring + 10.0 port

luc-demeyer committed:
bugfix in raise

luc-demeyer committed:
fix init flag when computing depreciation board

luc-demeyer committed:
hide fields for type view assets

luc-demeyer committed:
fix viexs

grindtildeath committed:
Fix compute methods dependencies and small optimizations

grindtildeath committed:
Rename demo file to test and move it into the right folder

luc-demeyer committed:
fix test data

luc-demeyer committed:
fix onchange

grindtildeath committed:
[MIG] account_asset_management: Migration to 11.0

grindtildeath committed:
Split README into parts

grindtildeath committed:
Fix tests

grindtildeath committed:
Fix asset and profile form view JS error

grindtildeath committed:
Cosmetic changes

grindtildeath committed:
Do not use builtin type in function

luc-demeyer committed:
fix _compute_move_line_check

luc-demeyer committed:
simplify domain

grindtildeath committed:
Fix KeyError when invl_id does not exist

grindtildeath committed:
Drop unsupported hierarchical tree view

grindtildeath committed:
Improvements according to review

grindtildeath committed:
Drop migrations script

grindtildeath committed:
Improve calls to super with Python3

grindtildeath committed:
Move from TransactionCase to SavepointCase

grindtildeath committed:
Second round of improvements

luc-demeyer committed:
fix local variable 'dlines' referenced before assignment

grindtildeath committed:
Fix use of Date vs Datetime for method_number_end

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update account_asset_management.pot

oca-travis committed:
[UPD] Update account_asset_management.pot

sbidoul committed:
[IMP] asset mgt: use exclude manifest key

This is now supported natively by Odoo 11.

oca-travis committed:
[UPD] Update account_asset_management.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

oca-travis committed:
[UPD] Update account_asset_management.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

norlinhenrik committed:
[MIG] account_asset_management: Migration to 12.0

ghost commented:
I have now signed both Individual CLA for @norlinhenrik and Entity CLA for @apps2grow.
I don't know why no github login is found for Stefan Rijnhard. He is not in my team...

pedrobaeza commented:
@apps2grow don't worry about that one, as it's due to old emails being imported on commit history. We are tuning the CLA bot for that. See https://github.com/OCA/oca-github-bot/issues/28 for more details.

ghost commented:
I would like to make a migration script for this module. I am learning by reading other migration scripts. But how to RUN the migration script?

ghost commented:
Could this module merged? I want to translate it in german fully.

tbaden commented:
@apps2grow 
to run a migration script you need a migrations directory, with a subdir with the new version_number of your module (e.g. 12.0.1.0.1). In there you can create your pre-migrate.py or post-migrate.py file. It'll run on module update, don't forget to set your version number correctly in the manifest.py

sbidoul commented:
@apps2grow thanks for this work.

Regarding the fiscal year, IMO the dependency on OCA `account_fiscal_year` can be removed, and the search on the fiscal year `date.range` can be replaced by a search on the standard Odoo model `account.fiscal.year`.

About `parent_path` is a migration script necessary? @pedrobaeza can you confirm?

sbidoul commented:
Or better, use `compute_fiscalyear_dates(self, current_date)` on res.company to find the fiscal year end date.

randall-vx commented:
Hi @pedrobaeza ! What do you think about @sbidoul comments ? Regards !

WhiteTiger08 commented:
Any progress  here? may I help with something?

MaxyMoos committed:
account_asset_management: Add unit test for invoices with multiple assets

MaxyMoos committed:
Included PR #828 Fix computation of depreciation lines when having multiple assets in invoice

pedrobaeza commented:
@apps2grow please include #828

ghost commented:
How do I do it? I am new to this, but I would like to learn :)

pedrobaeza commented:
You can use `git cherry-pick` with the hash of the commit in the PR. If several of them, just repeat the command with each.

luc-demeyer commented:
I agree with @sbidoul to move from date_range to account.fiscal.year.
Concerning parent_id/child_ids : my intention is to remove this completely and move to tags for reporting purposes. I think we can already remove this now and work on the reporting module for Odoo 12.0 later on (e.g. OCA code sprint  during Odoo Experience).
https://github.com/OCA/account-financial-tools/pull/828 should indeed be included.

WhiteTiger08 commented:
What else are you missing here?

norlinhenrik committed:
Merge branch '12.0' of https://github.com/oca/account-financial-tools into 12.0-mig-account_asset_management

Saran440 commented:
Functional test
Case 1: Asset created directly -> salvage value is editable -> OK
Case 2: Asset created by supplier invoice -> salvage value is not editable -> Not OK

kittiu commented:
I think many is expecting this important addons to be merged. Anything we can help?

carlosdauden commented:
Wrong depreciation value when Computation Method is Linear up to Salvage Value and Prorata Temporis cheked.

http://3374499-756-f4d651.runbot3.odoo-community.org/web#id=2&action=254&model=account.asset&view_type=form&menu_id=89

![Selección_233](https://user-images.githubusercontent.com/7701001/59373222-867d5300-8d49-11e9-9949-32073b482c1e.png)
![Selección_234](https://user-images.githubusercontent.com/7701001/59373245-909f5180-8d49-11e9-8334-4de9c65e9c2c.png)

Could be the problem here?:
https://github.com/OCA/account-financial-tools/blob/c507c7307639fed171196d5faac5d36af09c0bcc/account_asset_management/models/account_asset.py#L995-L1010

luc-demeyer commented:
@Saran440 
can you open an issue for the problem with the not editable salvage value ?
This issue has no relation to the 12.0 port (and needs to be fixed also on the previous version of this module.

@carlosdauden 
same remark as for issue reported by @Saran440

pedrobaeza commented:
@luc-demeyer but we can't merge this for being used in Spain instead of `account_asset` if it's not fixed...

luc-demeyer commented:
@kittiu 
Question on : can we help ?
If you have time:
- removal of dependency on OCA account_fiscalyear
- removal of 'view' assets with parent_id (confusing, I already intended to do it during the 11.0 port but didn't find the time yet). This parent assets of type view are there only for reporting purposes. I propose to replace them by an optional field for m2m reporting tags. These tags should have a parent_id.
I think we should remove the confusing assets of type view before merging the 12.0 port. Implementing the tag based reports can be done in a later stage (my idea is to do this during the october OCA code sprint).

luc-demeyer commented:
@pedrobaeza 
which one of the reported issues is critical for the spanish market and therefor blocking for a merge ?
In Belgium, I encountered the 'Linear up to Salvage Value' only once in the past 10 years. The issue reported by @Saran440 looks more important to me.

pedrobaeza commented:
@luc-demeyer prorata temporis is always used in Spain, but now I'm seeing that the compute method is not the standard one, so if @carlosdauden confirms that the problem doesn't happen with other methods, it's OK for me.

carlosdauden commented:
Ok.
I will open a new PR or Issue with that problem.

luc-demeyer commented:
@carlosdauden 
if you open an issue then I'll fix it (and add a unit test) but since I am leaving on holiday tomorrow morning it may take a while.

carlosdauden commented:
@luc-demeyer 
Issue opened.
This isn't too important, enjoy your holidays. :wink:

sbidoul commented:
So is it correct the last two things to do here are
- [ ] remove the dependency on account_fiscal_year (should be trivial)
- [ ] cherry pick #838

sbidoul commented:
@luc-demeyer IMO we should keep the view assets until the asset tags are implemented with a proper migration script from view assets to tags.

ghost commented:
I did cherry pick #828 already, see in the end of the commit history.

sbidoul commented on account_asset_management/__manifest__.py:
```diff
@@ -0,0 +1,30 @@
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'depends': [
+        'account_fiscal_year',
```
```suggestion
        'account',
```

sbidoul commented:
@apps2grow #828 yes, I was mentioning #838.

sbejaoui commented:
> last two things to do here are
> 
> * [ ]  remove the dependency on account_fiscal_year (should be trivial)
> * [ ]  cherry pick #838

Hi @apps2grow ,
I made a [PR](https://github.com/apps2grow/account-financial-tools/pull/1) to your source branch that resolve the two remaining tasks to finish this one.

Please consider review it and merge it if it looks good to you.

Best regards.

sbejaoui committed:
Finish 12.0 migration of account_asset_management

* account_asset: Do not loop on all the lines to search for one linked asset

Before this change, the use of `mapped` on self did loop on all the move
lines that are included in self to get the assets, what could be very
costly for a simple write on a lot of move lines. As the goal is to raise
an error only if at least one move is linked to an asset, we break the
loop if the condition is fulfilled.

* performance improvement

* [RMV] - Remove useless dependency

In 12.0 account_fiscal_year is a standard feature no need to depend on oca
module account_fiscal_year

ghost commented:
I have merged the PR from @sbejaoui

luc-demeyer commented:
> @luc-demeyer IMO we should keep the view assets until the asset tags are implemented with a proper migration script from view assets to tags.

@sbidoul 
I agree with the principle but that is a bit more work and keeps the confusing User Experience caused by the view assets until we implement the reporting tags (I can take the lead on this during the october OCA code sprint). If we go that way, than we should also migrate xls reporting module (and hence a request if you could merge https://github.com/OCA/account-financial-tools/pull/701). I'll do the migration of the reporting module to 12.0 shortly after the merge (it should be primarily a matter of testing, hence not a lot of work).

pedrobaeza commented:
I would like to have this merged ASAP, and of course the view assets are very confusing, but I can't push you to any path.

luc-demeyer commented on account_asset_management/models/__init__.py:
```diff
@@ -0,0 +1,9 @@
+from . import account_asset_recompute_trigger
+from . import account_invoice
+from . import account_move
+#from . import date_range
```
we should replace this import by "from . import account_fiscal_year", rename the date_range.py to account_fiscal_year.py and adjust the logic for table recompute when creating/changing fiscal year duration.

luc-demeyer commented on account_asset_management/models/account_asset.py:
```diff
@@ -0,0 +1,1097 @@
+    account_analytic_id = fields.Many2one(
+        comodel_name='account.analytic.account',
+        string='Analytic account',
+        domain=[('type', '!=', 'view'),
```
Fix https://github.com/OCA/account-financial-tools/commit/fc09774f7bf889b431e46f099d7d2eacdd6db498 to be cherrypicked.

norlinhenrik committed:
[12.0][FIX] account_asset_management: date_range -> account_fiscal_year, account_analytic_id without domain

ghost commented:
I have updated the pull request now, with update in account_fiscal year.py and fix in account_asset.py.

luc-demeyer commented:
@apps2grow 
Thanks a lot. I reviewed and tested the new code. Looks ok to me.
The only thing that needs to be done is to fix the few remaing flake8 issues and than we are ready for merge.

norlinhenrik committed:
[FIX] test_flake8

ghost commented:
Done.

I need to migrate from https://github.com/odoo/odoo/tree/10.0/addons/account_asset to https://github.com/OCA/account-financial-tools/tree/10.0/account_asset_management in version 10.0 or 11.0. Does anyone have a migration script for this?

norlinhenrik committed:
[FIX] test_flake8 (2)

sbidoul commented:
/ocabot merge

ghost commented:
Great! How to replace account_asset with account_asset_management?

luc-demeyer commented:
You'll have to develop a migration script.
You can have a look to https://github.com/OCA/account-financial-tools/tree/10.0/account_asset_management/migrations/8.0.3 and use this as a starting point, but it will be a bit of work to get a seemless migration from the odoo account_asset module to this one. I am also not sure if we want to have such a script as part of this module or rather have a separate module that does this.

pedrobaeza commented:
It must inside this module, @luc-demeyer. It doesn't affect for new installations and people migrating from already installed `account_asset_management`. It should be detected in the migration script itself (something like `if not openupgrade.table_exists(...): return`

luc-demeyer commented:
I assume we should use post_init_hook for this.

pedrobaeza commented:
No, we should use regular migration mechanism + a module merge in OpenUpgrade, which I have just made there: https://github.com/OCA/OpenUpgrade/blob/4f1b58559ef590cca00a28adc5dc23aada0bd7be/odoo/addons/openupgrade_records/lib/apriori.py#L44

sbidoul commented:
Shouldn't we provide a path for people who want to transition without a using openupgrade?

pedrobaeza commented:
@sbidoul the path is to run manually the scripts in their DB. What other path can be?

sbidoul commented:
post_init_hook as suggested by Luc? Could be dangerous though, I'm not sure.

pedrobaeza commented:
That's not feasible, as you come from `account_asset` module installed, so you mustn't install also `account_asset_management` and then uninstall the other: you will lose data and mix references. That's why the path is to move all references from `account_asset` to `account_asset_management` and then move corresponding data between tables. If you are not using OpenUpgrade, then you must do it manually, using `openupgradelib` methods for updating that name... in a shell session for example.

nikul-serpentcs commented:
IMP https://github.com/OCA/account-financial-tools/pull/852
